### PR TITLE
fix(developer): enforce presence of Version field when FollowKeyboardVersion is not set, in package compiler

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes: #12193

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-package/test/fixtures/absolute_path/source/absolute_path.kps [hunk 1]
```diff
@@ -17,6 +17,7 @@
   </StartMenu>
   <Info>
     <Name URL="">Absolute Path</Name>
+    <Version>1.0</Version>
```
Comment (mcdurdin) [inline developer/src/kmc-package/test/fixtures/absolute_path/source/absolute_path.kps:20]: This was technically wrong before

File: developer/src/kmc-package/test/test-package-compiler.ts [hunk 2]
```diff
@@ -209,6 +209,8 @@ describe('KmpCompiler', function () {
       kmpJson = kmpCompiler.transformKpsToKmpObject(kpsPath);
     });
 
+    assert.isNotNull(kmpJson);
```
Comment (mcdurdin) [inline developer/src/kmc-package/test/test-package-compiler.ts:212]: Verifying kmpJson is not null avoids a more mysterious error later in the test

**Comment by mcdurdin** [comment]
This may not be the right way to do this for stability reasons -- the keyboard info compiler already infers the version number '1.0' if it is not present in the package file (#11985), but this fix was not cherry-picked to stable-17.0.

Cherry-picking #11985 to stable-17.0 would mitigate the metadata problem.

However, inferring '1.0' is also not great because that could be confusing to keyboard authors.

@LornaSIL @DavidLRowe: would you prefer if the Package Compiler gave an error for a missing version number (where FollowKeyboardVersion is not in use), or if it just assumed '1.0' in those situations?

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.91-alpha](https://keyman.com/downloads/releases/alpha/18.0.91)

**Pull request merged**